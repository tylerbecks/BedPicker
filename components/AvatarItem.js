import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

const AvatarItem = ({ photo, text }) => (
  <View style={styles.item}>
    <Image
      style={styles.avatar}
      source={photo ? photo : require('../assets/images/jerry.png')}
    />
    <Text style={styles.itemText}>{text}</Text>
  </View>
);

export default AvatarItem;

const AVATAR_SIZE = 40;

const styles = StyleSheet.create({
  item: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingVertical: 8,
    marginLeft: 20
  },
  itemText: {
    fontSize: 14
  },
  avatar: {
    height: AVATAR_SIZE,
    width: AVATAR_SIZE,
    borderRadius: AVATAR_SIZE / 2,
    marginRight: 10
  }
});
