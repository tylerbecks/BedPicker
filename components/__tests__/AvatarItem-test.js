import 'react-native';
import React from 'react';
import AvatarItem from '../AvatarItem';
import renderer from 'react-test-renderer';

it('renders when guests have photos', () => {
  expect(
    renderer.create(<AvatarItem text="baz" photo={{}} />).toJSON()
  ).toMatchSnapshot();
});
