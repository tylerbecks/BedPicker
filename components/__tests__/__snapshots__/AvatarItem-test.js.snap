// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when guests have photos 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginLeft": 20,
      "paddingVertical": 8,
    }
  }
>
  <Image
    source={Object {}}
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "marginRight": 10,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    baz
  </Text>
</View>
`;
